File: SpringBootApp/src/main/java/com/example/springbootapp/data/dto/CartItemDto.java
package com.example.springbootapp.data.dto;

import com.example.springbootapp.data.entities.Embdeddables.Personalization;

public class CartItemDto {
    private String id;
    private String quantity;
    private Personalization personalization;
    private ProductWithVariantDto productWithVariant;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/CartItemDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.Personalization
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CartItemDto(
    val id: String,
    val quantity: Int,
    val personalization: Personalization?,
    val product: ProductWithVariantDto,
    val price: Double
)